src/main/java/com/prowidesoftware/swift/io/parser/SwiftParser.java

```diff
@@ -425,10 +430,18 @@ private SwiftBlock createBlock(final char blockId, final String s) {
                 b = createBlock1(s);
                 break;
             case '2': // block 2 (single valued)
-                if (isInput(s)) {
+                String block2Type = getBlock2Type(s);
+                if ("I".equals(block2Type)) {
                     b = createBlock2Input(s);
                 } else {
                     b = createBlock2Output(s);
+                    if (!"O".equals(block2Type)) {
+                        final String error = "The Block2 type could not be identified";
```

Comment:
"Expected an "I" or "O" to identify the block 2 type (direction) and found {0}"
Donde {0}" seria el char que parseo

Reply:
Ok